File: app/src/main/java/me/gabeg/sicksends/toprope/SsTopRopeViewModel.java
package me.gabeg.sicksends.toprope;

import android.app.Application;

import me.gabeg.sicksends.problem.SsProblemViewModel;

/**
 * Top rope view model.
 */
public class SsTopRopeViewModel
	extends SsProblemViewModel<SsTopRopeRepository, SsTopRope>
{

	/**
	 */
	public SsTopRopeViewModel(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemViewModel#buildRepository(Application)
	 */
	public void buildRepository(Application app)
	{
		this.mRepository = new SsTopRopeRepository(app);
	}

}

File: app/src/main/java/me/gabeg/sicksends/toprope/SsTopRopeViewModel.kt
package me.gabeg.sicksends.toprope

import androidx.lifecycle.SavedStateHandle
import dagger.hilt.android.lifecycle.HiltViewModel
import me.gabeg.sicksends.problem.SsGenericProblemViewModel
import javax.inject.Inject

/**
 * Top rope view model.
 */
@HiltViewModel
class SsTopRopeViewModel @Inject constructor(
	savedStateHandle: SavedStateHandle,
	repo : SsTopRopeRepository)
		: SsGenericProblemViewModel<SsTopRopeProblem>(savedStateHandle, repo)
